sourcecode/src-0/sourcecode/Macros.scala

```diff
@@ -0,0 +1,260 @@
+package sourcecode
+
+import scala.language.implicitConversions
+import scala.quoted._
+import scala.tasty.Reflection
+
+trait NameMacros {
+  inline implicit def generate: Name =
+    ${ Macros.nameImpl }
+}
+
+trait NameMachineMacros {
+  inline implicit def generate: Name.Machine =
+    ${ Macros.nameMachineImpl }
+}
+
+trait FullNameMacros {
+  inline implicit def generate: FullName =
+    ${ Macros.fullNameImpl }
+}
+
+trait FullNameMachineMacros {
+  inline implicit def generate: FullName.Machine =
+    ${ Macros.fullNameMachineImpl }
+}
+
+trait FileMacros {
+  inline implicit def generate: sourcecode.File =
+    ${ Macros.fileImpl }
+}
+
+trait FileNameMacros {
+  inline implicit def generate: sourcecode.FileName =
+    ${ Macros.fileNameImpl }
+}
+
+trait LineMacros {
+  inline implicit def generate: sourcecode.Line =
+    ${ Macros.lineImpl }
+}
+
+trait EnclosingMacros {
+  inline implicit def generate: Enclosing =
+    ${ Macros.enclosingImpl }
+}
+
+trait EnclosingMachineMacros {
+  inline implicit def generate: Enclosing.Machine =
+    ${ Macros.enclosingMachineImpl }
+}
+
+trait PkgMacros {
+  inline implicit def generate: Pkg =
+    ${ Macros.pkgImpl }
+}
+
+trait TextMacros {
+  inline implicit def generate[T](v: => T): Text[T] = ${ Macros.text('v) }
+  inline def apply[T](v: => T): Text[T] = ${ Macros.text('v) }
+}
+
+trait ArgsMacros {
+  inline implicit def generate: Args =
+    ${ Macros.argsImpl }
+}
+
+object Util{
+  def isSynthetic(c: Reflection)(s: c.Symbol) = isSyntheticName(getName(c)(s))
+  def isSyntheticName(name: String) = {
+    name == "<init>" || (name.startsWith("<local ") && name.endsWith(">"))
+  }
+  def getName(c: Reflection)(s: c.Symbol) = {
+    import c.given
+    s.name.trim
+      .stripSuffix("$") // meh
+  }
+}
+
+object Macros {
+
+  def actualOwner(c: Reflection)(owner: c.Symbol): c.Symbol = {
+    import c.given
+    var owner0 = owner
+    // second condition is meh
+    while(Util.isSynthetic(c)(owner0) || Util.getName(c)(owner0) == "ev") {
```

Comment:
simpler: `owner0.flags.is(Flags.Synthetic)`, but it may not work, as the semantics for `Util.isSynthetic` seem to be special.

Reply:
I tried it, didn't work